### ISSUE TITLE
Method to create tab components (CreateTextSegmentedControl) fail and cause crash.

### ISSUE THREAD

**Issue by IMB11** [darknight1050/questui#70]
QuestUI 0.17.5 for Beat Saber 1.25.1

```cpp
if (firstActivation) {
            l("activfirst")
            static ArrayW<StringW> options(3);
            l("createdArrayW")

            options[0] = "Controller";
            l("createdArrayW0")
            options[1] = "Gameplay";
            l("createdArrayW1")
            options[2] = "UI";
            l("createdArrayW2")
//            options[3] = "Miscellaneous";

            CreateTextSegmentedControl(this->get_transform(), Vector2(75, 7), options, [this](int PH1) {
                this->selectedTab = PH1;
                SwitchGameplayTab(PH1);
            });
            l("createdSegtab")
            // ...
}
```

Crashes at `CreateTextSegmentedControl`. 

```

12-23 21:54:04.820 23552 23604 I QuestHook[toomanytweaks|v0.2.0]: activfirst
12-23 21:54:04.820 23552 23604 I QuestHook[toomanytweaks|v0.2.0]: createdArrayW
12-23 21:54:04.820 23552 23604 I QuestHook[toomanytweaks|v0.2.0]: createdArrayW0
12-23 21:54:04.820 23552 23604 I QuestHook[toomanytweaks|v0.2.0]: createdArrayW1
12-23 21:54:04.820 23552 23604 I QuestHook[toomanytweaks|v0.2.0]: createdArrayW2
... tombstone stuff here

Dumped tombstone:
********** Crash dump: **********
Build fingerprint: 'oculus/hollywood/hollywood:10/QQ3A.200805.001/49068160412800150:user/release-keys'
#00 0x0000000000080a74 /apex/com.android.runtime/lib64/bionic/libc.so (prop_area::find_property(prop_bt*, char const*, unsigned int, char const*, unsigned int, bool)+68) (BuildId: 74793b832f73c27f2be0922f70cc8e16)
#01 0x000000000f8b37b4 /data/data/com.beatgames.beatsaber/files/libcodegen.so (std::get_unexpected()) (BuildId: 1c74ec3efe44d71e52f55b03528304b2fa65db11)
Crash dump is completed
```

**Comment by IMB11** [comment]
It seems this issue was fixed in 0.17.6 (https://github.com/darknight1050/QuestUI/releases/tag/v0.17.6) but is not available for 1.25.1?

**Comment by IMB11** [comment]
If you encounter this yourself, use the patched method:

```cpp
QuestUI::CustomTextSegmentedControlData* CreateTextSegmentedControl(UnityEngine::Transform* parent, UnityEngine::Vector2 anchoredPosition, UnityEngine::Vector2 sizeDelta, ArrayW<StringW> values, std::function<void(int)> onCellWithIdxClicked)
    {
        static SafePtrUnity<TextSegmentedControl> segmentedControlTemplate;
        if (!segmentedControlTemplate)
            segmentedControlTemplate = Resources::FindObjectsOfTypeAll<HMUI::TextSegmentedControl*>().FirstOrDefault([](auto x){
                return x->get_name() == "TextSegmentedControl";
        });

        auto segmentedControlObj = Object::Instantiate(segmentedControlTemplate->get_gameObject(), parent, false);
        segmentedControlObj->SetActive(false);
        static ConstString QuestUITextSegmentedControl("QuestUITextSegmentedControl");
        segmentedControlObj->set_name(QuestUITextSegmentedControl);
        auto rectTransform = reinterpret_cast<RectTransform*>(segmentedControlObj->get_transform());
        rectTransform->set_sizeDelta(sizeDelta);
        rectTransform->set_anchoredPosition(anchoredPosition);


        Object::DestroyImmediate(segmentedControlObj->GetComponent<HMUI::TextSegmentedControl*>());
        auto control = segmentedControlObj->AddComponent<HMUI::SegmentedControl*>();
        auto result = segmentedControlObj->AddComponent<QuestUI::CustomTextSegmentedControlData*>();
        
        result->firstCellPrefab = segmentedControlTemplate->firstCellPrefab;
        result->lastCellPrefab = segmentedControlTemplate->lastCellPrefab;
        result->middleCellPrefab = segmentedControlTemplate->middleCellPrefab;
        result->singleCellPrefab = segmentedControlTemplate->singleCellPrefab;

        result->segmentedControl = control;
        control->dataSource = reinterpret_cast<HMUI::SegmentedControl::IDataSource*>(result);
        
        if (onCellWithIdxClicked)
        {
            using DelegateType = System::Action_2<HMUI::SegmentedControl*, int>*;
            std::function<void(HMUI::SegmentedControl*, int)> fun = [onCellWithIdxClicked](HMUI::SegmentedControl* cell, int idx){ onCellWithIdxClicked(idx); };
            auto delegate = MakeDelegate(DelegateType, fun);
            control->add_didSelectCellEvent(delegate);
        }

        int childCount = result->get_transform()->get_childCount();
        for (int i = 0; i < childCount; i++) 
        {
            Object::DestroyImmediate(result->get_transform()->GetChild(0)->get_gameObject());
        }
        
        result->set_texts(values);

        segmentedControlObj->SetActive(true);
        return result;
    }
```
